File: Semana_2/Playwright/TypeScript/npm/curso/e2e/pages/classic/accounts-overview.page.ts
import { Locator, Page } from '@playwright/test';

export class AccountsOverviewPage {
  private readonly pageTitle: Locator;
  private readonly accountsTable: Locator;

  constructor(page: Page) {
    this.pageTitle = page.getByRole('heading', { name: 'Accounts Overview' });
    this.accountsTable = page.getByRole('table');
  }

  title(): Locator {
    return this.pageTitle;
  }

  table(): Locator {
    return this.accountsTable;
  }
}

File: Semana_2/Playwright/Java/maven/curso/e2e/src/test/java/com/curso/playwright/pages/classic/AccountsOverviewPage.java
package com.curso.playwright.pages.classic;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.microsoft.playwright.options.AriaRole;

public class AccountsOverviewPage {
    private final Locator title;
    private final Locator accountsTable;

    public AccountsOverviewPage(Page page) {
        this.title = page.getByRole(AriaRole.HEADING, new Page.GetByRoleOptions().setName("Accounts Overview"));
        this.accountsTable = page.getByRole(AriaRole.TABLE);
    }

    public String title() {
        title.waitFor();
        return title.innerText().trim();
    }

    public boolean hasAccountsTable() {
        accountsTable.waitFor();
        return accountsTable.isVisible();
    }
}
